### ISSUE TITLE
reproduce results on MSRVTT

### ISSUE THREAD

**Issue by hitexcalibur** [mayuelala/SimVTP#4]
Hi,

Great work and thanks for sharing the code!

I tried to run inference with your model checkpoint and code on MSRVTT but couldn't get the same top1 recall. I wonder if it's possible to also share your text and video embedding results on MSRVTT test set to help debug.

Also for the tokenizer and text embedding, the code loads from `/apdcephfs/private_mayuema/bert`, I wonder if this is the same as [bert-base-uncased](https://huggingface.co/bert-base-uncased) from hugging face.

**Comment by hitexcalibur** [comment]
After looking into the problem a bit closer, I noticed two potential reasons:

1. It seems that you are using the `miech` split of MSRVTT while a bunch of other papers uses `jsfusion`.
2. Looking at the json file you uploaded on Drive, seems that for each video, you concatenate the captions from multiple annotators which I'm not sure if it's a common practice on MSRVTT.

Please correct me if I made any mistake. and I wonder if you have any results using the jsfusion split and with one caption per video for both finetuning and testing on MSRVTT

**Comment by huchaodi** [comment]
用了这个测试集，原来recall@1只有 0.34,  直接到 0.62 ...

**Comment by SCZwangxiao** [comment]
> Hi,
> 
> Great work and thanks for sharing the code!
> 
> I tried to run inference with your model checkpoint and code on MSRVTT but couldn't get the same top1 recall. I wonder if it's possible to also share your text and video embedding results on MSRVTT test set to help debug.
> 
> Also for the tokenizer and text embedding, the code loads from `/apdcephfs/private_mayuema/bert`, I wonder if this is the same as [bert-base-uncased](https://huggingface.co/bert-base-uncased) from hugging face.

Does the gap very significant? I am going to re-train the model on MSRVTT-JSFusion.

**Comment by SCZwangxiao** [comment]
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/31362395/224644984-5510058e-e89a-44a6-8c1a-b1346fd1906b.png">

**Comment by hitexcalibur** [comment]
@SCZwangxiao Thanks for running this. The numbers on MSRVTT-JSFusion are close to what I have.

One question for your plot, for miech, do you use one caption per video or also concatenate multiple captions?

**Comment by SCZwangxiao** [comment]
> @SCZwangxiao Thanks for running this. The numbers on MSRVTT-JSFusion are close to what I have.
> 
> One question for your plot, for miech, do you use one caption per video or also concatenate multiple captions?

I concatenate multiple captions for miech (using the authors' annotation in Google Drive). This setting makes the task much easier.

**Comment by aiden-s-lee** [comment]
@SCZwangxiao Thank you for the clarification. In this case, would you be willing to change the reported number in the paper? It may be misleading for the readers in some cases.

**Comment by SCZwangxiao** [comment]
> 

Sorry, but I am not the author of the paper, lol. Therefore, I am unsure whether there is any difference between the experimental settings. Below are mine:

- I concluded experiments by using code in this repo, data from [Frozen](https://github.com/m-bain/frozen-in-time), and BERT tokenizer from [bert-base-uncased](https://huggingface.co/bert-base-uncased). I did not change anything in the code (except for the data and tokenizer path). 
- I conducted experiments in the NVIDIA DGX A100 cluster, detailed info is listed below:
```bash
------------------------------------------------------------
System environment:
    sys.platform: linux
    Python: 3.7.4 (default, Aug 13 2019, 20:35:49) [GCC 7.3.0]
    CUDA available: True
    numpy_random_seed: 1440163833
    GPU 0,1,2,3,4,5,6,7: NVIDIA A100-SXM4-40GB
    CUDA_HOME: /usr/local/cuda
    NVCC: Cuda compilation tools, release 11.7, V11.7.99
    GCC: gcc (GCC) 7.5.0
    PyTorch: 1.13.1+cu117
    PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.7
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.4.1  (built against CUDA 11.6)
    - Built with CuDNN 8.5
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.7, CUDNN_VERSION=8.5.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.13.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

    TorchVision: 0.14.1+cu117
    OpenCV: 4.6.0

Runtime environment:
    cudnn_benchmark: False
    mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
    dist_cfg: {'backend': 'nccl'}
    seed: None
    Distributed launcher: pytorch
    Distributed training: True
    GPU number: 8
------------------------------------------------------------
```

**Comment by adeil-liv-pool** [comment]
@SCZwangxiao Apologies. I thought you were the author :). Thanks for sharing the experiment result though.

**Comment by mayuelala** [comment]
sorry，everyone，I will fix this bug and change the reported number in the paper ~

**Comment by Soldelli** [comment]
Dear @mayuelala could you kindly update us on the correct test performance on MSR-VTT?

**Comment by wufeim** [comment]
Which bert should I use to replace this line `tokenizer = BertTokenizerFast.from_pretrained('/apdcephfs/private_mayuema/bert')`? `bert-base-uncased`?

**Comment by wufeim** [comment]
> > 
> 
> Sorry, but I am not the author of the paper, lol. Therefore, I am unsure whether there is any difference between the experimental settings. Below are mine:
> 
> * I concluded experiments by using code in this repo, data from [Frozen](https://github.com/m-bain/frozen-in-time), and BERT tokenizer from [bert-base-uncased](https://huggingface.co/bert-base-uncased). I did not change anything in the code (except for the data and tokenizer path).
> * I conducted experiments in the NVIDIA DGX A100 cluster, detailed info is listed below:
> 
> ```shell
> ------------------------------------------------------------
> System environment:
>     sys.platform: linux
>     Python: 3.7.4 (default, Aug 13 2019, 20:35:49) [GCC 7.3.0]
>     CUDA available: True
>     numpy_random_seed: 1440163833
>     GPU 0,1,2,3,4,5,6,7: NVIDIA A100-SXM4-40GB
>     CUDA_HOME: /usr/local/cuda
>     NVCC: Cuda compilation tools, release 11.7, V11.7.99
>     GCC: gcc (GCC) 7.5.0
>     PyTorch: 1.13.1+cu117
>     PyTorch compiling details: PyTorch built with:
>   - GCC 9.3
>   - C++ Version: 201402
>   - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
>   - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
>   - OpenMP 201511 (a.k.a. OpenMP 4.5)
>   - LAPACK is enabled (usually provided by MKL)
>   - NNPACK is enabled
>   - CPU capability usage: AVX2
>   - CUDA Runtime 11.7
>   - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
>   - CuDNN 8.4.1  (built against CUDA 11.6)
>     - Built with CuDNN 8.5
>   - Magma 2.6.1
>   - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.7, CUDNN_VERSION=8.5.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.13.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 
> 
>     TorchVision: 0.14.1+cu117
>     OpenCV: 4.6.0
> 
> Runtime environment:
>     cudnn_benchmark: False
>     mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
>     dist_cfg: {'backend': 'nccl'}
>     seed: None
>     Distributed launcher: pytorch
>     Distributed training: True
>     GPU number: 8
> ------------------------------------------------------------
> ```

Hi @SCZwangxiao , may I ask how long does it take to reproduce the results on MSRVTT with an 8-GPU machine?
